Added functional tests for uncovered DeltaSpike quickstarts

rsmeral opened the pull request:
Depends on updated Graphene and Drone version in wfk bom https://github.com/jboss-developer/jboss-wfk-boms/pull/3

Commit:
Added functional tests for uncovered DeltaSpike quickstarts

Depends on updated Graphene and Drone version in wfk bom https://github.com/jboss-developer/jboss-wfk-boms/pull/3

sgilda commented:
I don't see any updated instructions in the README files for the quickstarts impacted by this update. Could you add a section with instructions on how to run the functional tests, similar to the ones added in the following commit? 
https://github.com/maschmid/jboss-wfk-quickstarts/commit/4f22b627466c6716d07a4073de11f5e2f3b7b01b

Commit:
Updated READMEs for functional tests of DeltaSpike QS

rsmeral commented:
Thanks for the reminder, I totally forgot.

sgilda commented:
Thanks @rsmeral . :-)

Commit:
Fix project and BOM versions

Commit:
Remove resource-plugin usage from deltaspike-authorization
* the resource-plugin execution configured the application user in AS by copying files directly to AS, so it would only work in the managed mode
* this is not really necessary because it is described in the README anyway, it just made testing easier

pmuir commented:
Code review passes, this is great stuff :-)

I tested the various tests, and I get errors on deltaspike authorization:

```
Failed tests:   loginTest(org.jboss.wfk.test.deltaspike.authorization.AuthorizationTest)

Tests in error: 
  employeeTest(org.jboss.wfk.test.deltaspike.authorization.AuthorizationTest): Cannot locate element using: input[value='Employees only Method']
  employeeTest(org.jboss.wfk.test.deltaspike.authorization.AuthorizationTest): Cannot locate element using: input[value='Logout']
  loginTest(org.jboss.wfk.test.deltaspike.authorization.AuthorizationTest): Cannot locate element using: input[value='Logout']
  adminTest(org.jboss.wfk.test.deltaspike.authorization.AuthorizationTest): Cannot locate element using: input[value='Admins only Method']
  adminTest(org.jboss.wfk.test.deltaspike.authorization.AuthorizationTest): Cannot locate element using: input[value='Logout']
```

Once this is fixed, I can merge this.

sgilda commented:
When I test deltaspike-authorization, I get a different result. 

[INFO] No tests to run.
[INFO] Surefire report directory: /home/sgilda/GitRepos/jboss-wfk-quickstarts/deltaspike-authorization/target/surefire-reports

---

##  T E S T S

Results :

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

Looking into it further.

sgilda commented on deltaspike-authorization/README.md:
```diff
@@ -83,6 +83,26 @@ Undeploy the Archive
+
+        mvn clean verify -Parq-jbossas-remote
+
+5. If you prefer to run the functional tests using managed instance of the JBoss server, meaning the tests will start the server for you, type fhe following command:
```
When they run the tests using a managed instance of the server, don't they need to configure the `jbossHome` property in the arquillian.xml file to point to the server path? The instructions are in the root README file, but you might want to point them there with something like:

_NOTE: The following commands assume you have configured your Maven user settings. If you have not, you must include Maven setting arguments on the command line. See [Run the Arquillian Tests](../README.md#run-the-arquillian-tests) for complete instructions and additional options._

sgilda commented on deltaspike-beanmanagerprovider/README.md:
```diff
@@ -82,6 +82,27 @@ Undeploy the Archive
+
+        mvn clean verify -Parq-jbossas-remote
+
+5. If you prefer to run the functional tests using managed instance of the JBoss server, meaning the tests will start the server for you, type fhe following command:
```
Same issue as above and for all the readme files.

Commit:
Fix password in deltaspike-authorization ftest

Commit:
Added note about JBOSS_HOME configuration for arq managed container + typo fix

rsmeral commented:
@pmuir Fixed, thanks. Wrong password was used.

@sgilda That happens when you try to execute the test goal on the quickstart project. The test needs to be run in the functional-tests project. It is separated this way to avoid running the functional test by accident and to emphasize that the functional tests are something that needs special conditions (a server, a browser and an already compiled artifact to test).

rsmeral commented:
@sgilda re:README: I updated the README as you suggested. Also, I checked the "Run the Arquillian tests" section which the note points to and updated that as well (https://github.com/jboss-developer/jboss-wfk-quickstarts/pull/16). WDYT?

rsmeral commented:
Also, if the README fix here is accepted, please fix also https://github.com/jboss-developer/jboss-wfk-quickstarts/blob/master/kitchensink-deltaspike/README.md. Thanks.

pmuir commented:
Merged.
